build-tools/packages/build-cli/src/library/repoPolicyCheck/fluidBuildTasks.ts

```diff
@@ -577,56 +644,40 @@ export const handlers: Handler[] = [
 		},
 	},
 	{
-		name: "fluid-build-tasks-tsc",
+		/**
+		 * Checks that all tsc project files (tsconfig.json), are only used once as the main
+		 * configuration among scripts.
+		 * Multiple uses may indicate a collision during build.
+		 */
+		name: "tsc-project-single-use",
 		match,
 		handler: async (file: string, root: string): Promise<string | undefined> => {
-			let json: PackageJson;
-			try {
-				json = JSON.parse(readFile(file)) as PackageJson;
-			} catch {
-				return `Error parsing JSON file: ${file}`;
-			}
-
-			if (!isFluidBuildEnabled(root, json)) {
-				return;
-			}
-			if (json.scripts === undefined) {
-				return;
-			}
-			const packageDir = path.dirname(file);
-			const errors: string[] = [];
-			const deps = getDefaultTscTaskDependencies(root, json);
-			const ignore = getFluidBuildTasksTscIgnore(root);
-			for (const [script, scriptCommands] of Object.entries(json.scripts)) {
-				if (scriptCommands === undefined) {
-					continue;
-				}
-				for (const commandUntrimmed of scriptCommands.split("&&")) {
-					const command = commandUntrimmed.trim();
-					if (!shouldProcessScriptForTsc(script, command, ignore)) {
-						continue;
+			const projectMap = new Map<string, string>();
+			// Note: this does check tsc-multi commands which do very likely reuse project files
```

Comment:
It does or doesn't check tsc-multi configs?

Reply:
It does not. But that is okay because we don't have any anymore. I'll be posting a tsc-multi cruft clean up PR shortly.

Reply:
Oh I see - bad comment...